lms/djangoapps/completion_api/tests/test_serializers.py

```diff
@@ -0,0 +1,101 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+
+from opaque_keys.edx.keys import CourseKey
+from progress import models
+
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase
+from xmodule.modulestore.tests.factories import ToyCourseFactory
+from ..serializers import StudentProgressSerializer
+from .. import views
+
+User = get_user_model()
+
+class StudentProgressSerializerTestCase(TestCase):
+
+    def test_serialize_student_progress_object(self):
+        test_user = User.objects.create(
+            username='test_user',
+            email='test_user@example.com'
+        )
+        progress = models.StudentProgress.objects.create(
+            user=test_user,
+            course_id=CourseKey.from_string('course-v1:abc+def+ghi'),
+            completions=17,
+        )
+        # annotate possible and percent
+        progress.possible = 20.0
+        progress.percent = 85
+        serial = StudentProgressSerializer(progress)
+        self.assertEqual(
+            serial.data,
+            {
+                'course_key': 'course-v1:abc+def+ghi',
+                'completion': {
+                    'earned': 17.0,
+                    'possible': 20.0,
```

Comment:
Very minor nit: 17/19 might be a better earned/possible pair, in order to test rounding - because 17/20 is exactly 85%